src/domain/image/download.ts

```diff
@@ -0,0 +1,39 @@
+
+import ImageData from './data/ImageData';
+import { UnsupportedContentSize, UnsupportedMimeType } from './errors';
+import save from './save';
+
+async function checkImage(picture: string): Promise<void>
+{
+    const response = await fetch(picture, { method: 'HEAD' });
+    const mimeType = response.headers.get('content-type');
+
+    if (mimeType === null || ['image/gif', 'image/jpeg', 'image/png'].includes(mimeType) === false)
+    {
+        throw new UnsupportedMimeType;
+    }
+
+    const length = response.headers.get('content-length');
+    const size = Number(length);
+
+    if (length === null || size > 1024 * 1024 * 5)
+    {
+        throw new UnsupportedContentSize;
+    }
+}
+
+export default async function download(picture: string, imageType: string): Promise<ImageData>
```

Comment:
The implementation of this function is not finished yet, but I've made a rough draft for you.

Reply:
Finished function